projects/storefrontlib/cms-components/misc/icon/icon-loader.service.ts

```diff
@@ -23,69 +24,76 @@ export class IconLoaderService {
 
   /**
    * Returns an html fragment which can be added to the DOM in a safe way.
+   *
+   * @deprecated only exists for backwards compatibility, html generation has been moved to icon component
    */
-  getHtml(type: ICON_TYPE | string): SafeHtml | undefined {
-    if (this.isResourceType(type, IconResourceType.SVG)) {
-      return this.sanitizer.bypassSecurityTrustHtml(
-        `<svg><use xlink:href="${this.getSvgPath(type)}"></use></svg>`
+  public getHtml(type: ICON_TYPE_STRING): SafeHtml | undefined {
```

Comment:
If `getHtml()` is not needed for backward compatibility with custom code, it can be removed.